cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,567 @@
+import itertools
+from typing import Sequence, Dict
+from cirq import circuits, devices, ops, protocols, sim, study, value
+import numpy
+from matplotlib import pyplot
+from mpl_toolkits.mplot3d import Axes3D
+
+
+class RabiExperimentResult(object):
+    """Class for storing and plotting data from a Rabi oscillation
+    experiment.
+    """
+
+    def __init__(self, num_shots: int, rabi_angles: Sequence[float],
+                 excited_state_pops: Sequence[float]):
+        """
+        Args:
+            num_shots: The number of repetitions of the circuit for each Rabi
+              angle.
+            rabi_angles: The rotation angles of the qubit around the x-axis
+              of the Bloch sphere.
+            excited_state_pops: The corresponding probabilities that the
+            qubit is in the excited state.
+        """
+        self._num_shots = num_shots
+        self._rabi_angles = rabi_angles
+        self._excited_state_pops = excited_state_pops
+
+    @property
+    def data(self) -> Dict[str, Sequence[float]]:
+        return {'rabi_angles': self._rabi_angles,
+                'excited_state_probability': self._excited_state_pops}
+
+    def plot(self) -> None:
+        fig = pyplot.figure()
+        pyplot.plot(self._rabi_angles, self._excited_state_pops, 'ro-',
+                    figure=fig)
+        pyplot.xlabel(r"Rabi Angle ($\pi$)")
+        pyplot.ylabel('Excited State Probability')
+
+
+class RandomizedBenchmarkingResult(object):
+    """Class for storing and plotting data from a randomized benchmarking
+    experiment.
+    """
+
+    def __init__(self, num_cfds_seq: Sequence[float],
+                 gnd_state_pops: Sequence[float]):
+        """
+        Args:
+            num_cfds_seq: The different numbers of Cliffords in the RB study.
+            gnd_state_pops: The corresponding average ground state
+              probabilities.
+        """
+        self._num_cfds_seq = num_cfds_seq
+        self._gnd_state_pops = gnd_state_pops
+
+    @property
+    def data(self) -> Dict[str, Sequence[float]]:
+        return {'number_of_cliffords': self._num_cfds_seq,
```

Comment:
Perhaps map number of cliffords directly to ground state probability? Again don't want users to have to memory strings. Could also define a `NamedTuple` type to return, because then IDEs will autocomplete.

Reply:
Done.